src/radix_connect/app_link_parser.rs

```diff
@@ -0,0 +1,154 @@
+use crate::prelude::*;
+use url::form_urlencoded;
+use url::Url;
+
+use super::{DappRequest, LinkRequest, MobileConnectRequest};
+
+#[uniffi::export]
+pub fn parse_request_url(url: String) -> Result<MobileConnectRequest> {
+    let connect_url = Url::parse(CONNECT_URL).unwrap();
+    match Url::parse(url.as_str()) {
+        Ok(parsed_url) => {
+            if parsed_url.host_str() != connect_url.host_str()
+                || parsed_url.scheme() != connect_url.scheme()
+            {
+                return Err(CommonError::RadixConnectMobileInvalidRequestUrl);
+            }
+
+            let query_parameters = parsed_url
+                .query_pairs()
+                .into_owned()
+                .collect::<HashMap<String, String>>();
+            if let Some(session_id) =
+                query_parameters.get(CONNECT_URL_PARAM_SESSION_ID)
+            {
+                match query_parameters.get(CONNECT_URL_PARAM_ORIGIN) {
+                    Some(origin) => {
+                        if let Ok(url) = Url::parse(origin) {
```

Comment:
use `parse_url` and return early...:

```rust
let url = parse_url(origin).map_err(|_| CommonError::RadixConnectMobileInvalidOrigin { bad_value: origin })?;
```

Reply:
I return early now but was not able to add `parse_url` so far

Reply:
what do you mean by "was not able to add `parse_url`" ? :) did you make it `pub fn`? if you add a `pub fn` into e.g. `string_utils.rs` and as long as you always in the top of each file do `use crate::prelude::*;` it should be visible :) 

the advantage of that small function is that it returns `Result<Url>` which just is an alias for `Result<Url, CommonError>` - which has the advatange of us being able in other functions returning `Result<T>` we can use just `parse_url(url)?;` to return early, **no need** to map the error type, since it is our own `CommonError` :) 

Ofc in some cases we *might* wanna return a specific error, relating to radix mobile connect. but maybe not?